hibernate-core/src/main/java/org/hibernate/procedure/internal/PostgresCallableStatementSupport.java

```diff
@@ -28,84 +31,59 @@
 	public JdbcCall interpretCall(
 			String procedureName,
 			FunctionReturnImpl functionReturn,
-			ParameterMetadataImplementor parameterMetadata,
+			ProcedureParameterMetadataImplementor parameterMetadata,
 			ProcedureParamBindings paramBindings,
 			SharedSessionContractImplementor session) {
-		throw new NotYetImplementedFor6Exception( getClass() );
+		final boolean firstParamIsRefCursor = parameterMetadata.getParameterCount() != 0
+				&& isFirstParameterModeRefCursor( parameterMetadata );
 
-//		// if there are any parameters, see if the first is REF_CURSOR
-//		final boolean firstParamIsRefCursor = ! parameterRegistrations.isEmpty()
-//				&& parameterRegistrations.get( 0 ).getMode() == ParameterMode.REF_CURSOR;
-//
-//		if ( firstParamIsRefCursor ) {
-//			// validate that the parameter strategy is positional (cannot mix, and REF_CURSOR is inherently positional)
-//			if ( parameterStrategy == ParameterStrategy.NAMED ) {
-//				throw new HibernateException( "Cannot mix named parameters and REF_CURSOR parameter on PostgreSQL" );
-//			}
-//		}
-//
-//		final StringBuilder buffer;
-//		if ( firstParamIsRefCursor ) {
-//			buffer = new StringBuilder().append( "{? = call " );
-//		}
-//		else {
-//			buffer = new StringBuilder().append( "{call " );
-//		}
-//
-//		buffer.append( procedureName ).append( "(" );
-//
-//		String sep = "";
-//
-//		// skip the first registration if it was a REF_CURSOR
-//		final int startIndex = firstParamIsRefCursor ? 1 : 0;
-//		for ( int i = startIndex; i < parameterRegistrations.size(); i++ ) {
-//			final ParameterRegistrationImplementor parameter = parameterRegistrations.get( i );
-//
-//			// any additional REF_CURSOR parameter registrations are an error
-//			if ( parameter.getMode() == ParameterMode.REF_CURSOR ) {
-//				throw new HibernateException( "PostgreSQL supports only one REF_CURSOR parameter, but multiple were registered" );
-//			}
-//
-//			for ( int ignored : parameter.getSqlTypes() ) {
-//				buffer.append( sep ).append( "?" );
-//				sep = ",";
-//			}
-//		}
-//
-//		return buffer.append( ")}" ).toString();
-	}
+		if ( firstParamIsRefCursor ) {
+			// validate that the parameter strategy is positional (cannot mix, and REF_CURSOR is inherently positional)
+			if ( parameterMetadata.hasNamedParameters() ) {
+				throw new HibernateException( "Cannot mix named parameters and REF_CURSOR parameter on PostgreSQL" );
+			}
+		}
 
-	@Override
-	public void registerParameters(
-			String procedureName,
-			CallableStatement statement,
-			ParameterStrategy parameterStrategy,
-			ParameterMetadataImplementor parameterMetadata,
-			SharedSessionContractImplementor session) {
-		throw new NotYetImplementedFor6Exception( getClass() );
+		final StringBuilder buffer;
+		if ( firstParamIsRefCursor ) {
+			buffer = new StringBuilder().append( "{? = call " );
+		}
+		else {
+			buffer = new StringBuilder().append( "{call " );
```

Comment:
Using the perfect size estimate would be awesome :)

```suggestion
			buffer = new StringBuilder(9 + procedureName.length() + registrations.size() * 2).append( "{call " );
```

Reply:
done, Thanks!